auth/handler.go

```diff
@@ -0,0 +1,68 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-dataset-api/permissions"
+	"github.com/ONSdigital/go-ns/common"
+	"github.com/ONSdigital/log.go/log"
+)
+
+//go:generate moq -out authtest/auth_mocks.go -pkg authtest . PermissionAuthenticator
+
+const (
+	CollectionIDHeader = "Collection-Id"
+	DatasetIDParam     = "dataset_id"
+)
+
+var (
+	getRequestVars func(r *http.Request) map[string]string
+	authenticator  PermissionAuthenticator
+)
+
+// Init set up function for the auth pkg. Requires a function for getting request parameters and a
+// PermissionsAuthenticator implementation
+func Init(GetRequestVarsFunc func(r *http.Request) map[string]string, PermissionsAuthenticator PermissionAuthenticator) {
```

Comment:
`init` function conveys a specific meaning about being implicitely called and taking no arguments in Go convention - i appreciate that as an exported function this is different, but I wonder if a different name would be clearer and avoid the conflation of the two concepts?

Admittedly, naming this is hard and I'm not sure what my suggestion would be. Maybe Create or Configure? `auth.Configure(funkything, implementationofinterface)` feels descriptive? (tbh, im not close enough to the code to understand why getRequestVars needs to be variable, to understand the full implications of what is being set by this function)

Reply:
renamed to Configure.